src/pkg/path/service_resource.go

```diff
@@ -0,0 +1,114 @@
+package path
+
+import (
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/common/str"
+	"github.com/alcionai/corso/src/internal/common/tform"
+)
+
+// ---------------------------------------------------------------------------
+// Tuple
+// ---------------------------------------------------------------------------
+
+// ServiceResource holds a service + resource tuple.  The tuple implies
+// that the resource owns some data in the given service.
+type ServiceResource struct {
+	Service           ServiceType
+	ProtectedResource string
+}
```

Comment:
I'm a bit on the fence about this struct definition and some of the validation that occurs for `[]ServiceResource`s. The below are the main concerns I have around this (numbered for ease of reference, not importance):
1. all fields in the struct are public, is there a time when callers would ever choose to use `MakeServiceResource` instead of declaring it themselves?
2. what, if any, guarantees are we making around validity of a `ServiceResource` instance, especially since all fields are public? Does this mean it'll be possible for code to generate and pass around invalid instances? Is all code that interacts with a `ServiceResource` now responsible for ensuring that instance is valid in some way?
3. same as (2) but this time for `[]ServiceResource` and around checking that subservice(s) are valid for the ordering of the `ServiceResource`s in the slice

To draw some parallels, the strictness of the `Path` initializer provides guarantees about individual `Path` instances. If some part of the code is handed a `Path` instance then that code can assume that at least the `Path` has some basic characteristics (i.e. non-empty tenant and resource owner, valid `ServiceType`/`CategoryType`, at least one non-prefix segment)

I'm happy to discuss this offline if that would be simpler